omniqueue/src/backends/redis/mod.rs

```diff
@@ -90,6 +90,16 @@ impl RedisConnection for RedisClusterConnectionManager {
 // the message has previously been received.
 struct InternalPayload<'a>(&'a [u8], usize);
 
+impl<'a> InternalPayload<'a> {
```

Comment:
I kind wish we'd just change `InternalPayload` to a struct with named fields, e.g.

```
struct InternalPayload<'a> {
    payload: &'a [u8],
    num_receives: usize,
}
```

Instead of adding method definitions to document the fields of a tuple struct.

Reply:
That's fair, but this type is instantiated everywhere and I don't like the additional verbosity. What if I just remove the helper methods i added?

Reply:
My concern is that if I'm just looking at the definition of `InternalPayload(&[u8], usize)`, the fields are difficult to understand. I can maybe infer the 0th field is the payload, but the meaning of the usize field is completely opaque (and easy to assume it has a different meaning than `num_receives`).

The methods you added definitely clarify what these fields mean, but I think it's a code smell that `InternalPayload`'s fields just need to be named explicitly. Code that's referencing this struct will be easier to understand, even if it's a bit more verbose.

Reply:
Alright, done. It's probably actually _less_ verbose now :smirk: